ui/app/components/clients/dashboard.js

```diff
@@ -190,10 +207,26 @@ export default class Dashboard extends Component {
   }
 
   @action
-  handleEditStartMonth() {
-    // ARG TODO will need to handle the action when the click save on the modal and you are given a new start Month, via month and year in params,
-    // if no endDate selected, default to 12 months range
-    // then send range to queryData this.queryData(range)
+  async handleClientActivityQuery(month, year, dateType) {
+    // dateType is either startTime or endTime
+    let monthIndex = this.arrayOfMonths.indexOf(month);
+    if (dateType === 'startTime') {
+      this.startTime = formatRFC3339(new Date(year, monthIndex));
+      this.endTime = null;
+    }
+    if (dateType === 'endTime') {
+      // this month comes in as an index
+      this.endTime = formatRFC3339(new Date(year, month));
+    }
+    try {
+      let response = await this.adapter.queryClientActivity(this.startTime, this.endTime);
+      // resets the endTime to what is returned on the response
+      this.endTime = response.data.end_time;
+      return response;
```

Comment:
This is the response from the new query. It's this data we want to pass into the charts components.